### PR TITLE
fix: combobox options exception on null values

### LINKED ISSUE #5458: ClrOptionItems cannot bind to a list with objects that have a 'null' property.
## Describe the bug

_When you bind a combobox to a list of options, an exception is thrown if any of the objects have a property with value null._

## How to reproduce

Stackblitz is down. In order to reproduce locally, in the Angular app:

1. Open _combobox.demo.ts_
2. Add a property 'test' with null value to the states list:

```
states = [
      {
          name: 'Alabama',
          abbreviation: 'AL',
          test: null
       },
...
```

3. See errors. Page does not render.

```
core.js:4610 ERROR TypeError: Cannot read property 'toString' of null
    at clr-angular.js:9732
    at Array.filter (<anonymous>)
    at clr-angular.js:9731
    at Array.filter (<anonymous>)
    at ClrOptionItems.updateItems (clr-angular.js:9727)
    at ClrOptionItems.set rawItems [as rawItems] (clr-angular.js:9704)
    at setInputsForProperty (core.js:10035)
    at elementPropertyInternal (core.js:9079)
    at Module.ɵɵproperty (core.js:15103)
    at ComboboxDemo_Template (combobox.demo.html:34)
```

## Expected behavior

Smart iterator ignores properties other than 'field name'.

## Versions

**App**

- Angular: 11
- Node: 12.16.0
- Clarity: tested on 4.0.7, 4.0.9 and 5.0xx (next)

**Device:**

- Type: Dell XPS 17
- OS: W10
- Browser Chrome
- Version 87

## Additional notes

Temporary fix: Map your list of options to a bare minimum list, before assigning.

### PR DESCRIPTION
**Pull request by Jinnie**
closes #5458
Signed-off-by: Ivan Donchev <idonchev@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Exception and broken component if any value in the object data is null

Issue Number: #5458

## What is the new behavior?

Handles null values by excluding them from filtering logic.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts [hunk 1]
```diff
@@ -72,9 +72,7 @@ export class ClrOptionItems<T> implements DoCheck, OnDestroy {
     if (this._filterField) {
       this.filteredItems = this._rawItems.filter(item => {
         const objValue = (item as any)[this._filterField];
-        return objValue !== undefined
-          ? objValue.toString().toLowerCase().indexOf(this.filter.toLowerCase().toString()) > -1
-          : false;
+        return objValue ? objValue.toString().toLowerCase().indexOf(this.filter.toLowerCase().toString()) > -1 : false;
```
Comment (bdryanovski) [inline packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts:75]: It could be simplified a bit

```ts
return  ('' + objValue).toLowerCase().includes(this.filter.toLowerCase())
```

File: packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts [hunk 2]
```diff
@@ -83,7 +81,7 @@ export class ClrOptionItems<T> implements DoCheck, OnDestroy {
           return item.toString().toLowerCase().indexOf(this.filter.toString().toLowerCase()) > -1;
         }
         const objValues = Object.values(item).filter(value => {
-          return value.toString().toLowerCase().indexOf(this.filter.toString().toLowerCase()) > -1;
+          return value ? value.toString().toLowerCase().indexOf(this.filter.toString().toLowerCase()) > -1 : false;
```
Comment (bdryanovski) [inline packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts:84]: Same here

```ts
return  ('' + objValue).toLowerCase().includes(this.filter.toLowerCase())
```

**Review by bdryanovski — APPROVED** [review]
Except that it could be simplified it looks fine to me.

File: packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts (hunk 1 above)
Comment (Jinnie) [reply packages/angular/projects/clr-angular/src/forms/combobox/option-items.directive.ts:75]: Thank. I guess it will work, but I want to explicitly exclude the null/undefined cases. I find it more readable and easier to maintain.

**Review by bbogdanov — APPROVED** [review]
